### PR TITLE
Update `DataCombined` and `SensitivityAnalysisResults` tests

### PR DESCRIPTION
**Pull request by IndrajeetPatil**
- remove redundant tests for `DataCombined`
- more validation for `SensitivityAnalysisResults`

### CODE REVIEW

**Comment by IndrajeetPatil** [comment]
Build failure due to:

```
The install of pandoc was NOT successful.
pandoc not installed. An error occurred during installation:
 The remote server returned an error: (429) Too Many Requests. Too Many Requests
```

**Review by msevestre — COMMENTED** [review]
Make the code readable unless what I am suggesting does not make sense

File: tests/testthat/test-data-combined.R [hunk 1]
```diff
@@ -92,12 +92,12 @@ if (.Platform$OS.type == "windows") {
...
   })
 
-  test_that("data frame dimensions are as expected when only `DataSet` is provided", {
+  test_that("data frame rows are as expected when only `DataSet` is provided", {
     myCombDat <- DataCombined$new()
     myCombDat$addDataSets(dataSet[[1]])
     df <- myCombDat$toDataFrame()
 
-    expect_equal(dim(df), c(12L, 16L))
+    expect_equal(dim(df)[[1]], 12L)
```
[the remark this answers is not in the document]
Comment (IndrajeetPatil) [reply tests/testthat/test-data-combined.R]: Definitely! Much more readable to use `nrow()`. Thanks. Done!

**Comment by IndrajeetPatil** [comment]
@msevestre This should be all set.

**Review by msevestre — APPROVED** [review]

**Pull request merged**